File: tests/fixtures/sample_elixir/sample_module.ex
defmodule SampleApp.Utils do
  @doc "Formats a greeting message"
  def greet(name) do
    "Hello, #{name}!"
  end

  @doc "Computes the sum of two numbers"
  def add(a, b) do
    a + b
  end

  defp private_helper(x) do
    x * 2
  end
end

defprotocol SampleApp.Serializable do
  def serialize(data)
end

defmodule SampleApp.Worker do
  def run(items) do
    SampleApp.Utils.greet("world")
    SampleApp.Utils.add(1, 2)
    process(items)
  end

  defp process(items) do
    Enum.map(items, fn x -> x end)
  end
end

File: tests/fixtures/sample_lua/sample_module.lua
-- Lua sample module for CodeAtlas tests

local M = {}

-- Format a greeting message
function M.greet(name)
    return "Hello, " .. name .. "!"
end

-- Compute the sum of two numbers
function M.add(a, b)
    return a + b
end

-- Top-level helper function
function process(items)
    M.greet("world")
    M.add(1, 2)
end

-- Local helper function
local function helper(x)
    return x * 2
end

-- Variable assigned to a function
local transform = function(data)
    return helper(data)
end

-- Configuration table (not a function)
local config = {
    debug = false,
    version = "1.0.0",
}

return M
